modules/sc-mesh-secure-deployment/src/nats/tests/test_cbma_adaptation.py

```diff
@@ -0,0 +1,505 @@
+        self.patcher_get_mac_addr.start()
+        result = self.cbma_adaptation.setup_cbma()
+        self.assertTrue(result)
+
+    def test_stop_cbma(self):
+        result = self.cbma_adaptation.stop_cbma()
+        self.assertTrue(result)
+
+    def test_setup__and_stop_cbma2(self):
+        # Tries to setup and stop CBMA with such config
+        # that doesn't have VLAN definitions.
+        self.cbma_adaptation2 = CBMAAdaptation(
+            self.mock_comms_ctrl, self.logger, self.mock_lock, "ms_config2.yaml"
+        )
+        # Allow "dummy" interfaces during unit testing
+        self.cbma_adaptation2.ALLOWED_KIND_LIST.add("dummy")
+
+        setup_result = self.cbma_adaptation2.setup_cbma()
+        self.assertTrue(setup_result)
+        stop_result = self.cbma_adaptation2.stop_cbma()
+        self.assertTrue(stop_result)
+
+    def test_reading_invalid_config_file(self):
+        mock_logger = MagicMock()
+
+        with patch.object(mock_logger, "getChild"):
+            # Mock the logger object returned by getLogger
+            mock_logger.getChild.return_value.error = MagicMock()
+
+            self.cbma_adaptation3 = CBMAAdaptation(
+                self.mock_comms_ctrl,
+                mock_logger,
+                self.mock_lock,
+                "invalid_yaml_file.yaml",
+            )
+            call_args_list = mock_logger.getChild.return_value.error.mock_calls
+
+            # Assert that the error method of the mocked logger.getChild()
+            # was called with the expected message
+            expected_message = "Error reading config file"
+            self.assertTrue(
+                any(
+                    expected_message in call_args[1][0] for call_args in call_args_list
+                ),
+                f"Expected message '{expected_message}' not found in logger calls.",
+            )
+            self.assertIsNone(self.cbma_adaptation3._CBMAAdaptation__config)
+            self.assertIsNone(self.cbma_adaptation3._CBMAAdaptation__cbma_config)
+            self.assertIsNone(self.cbma_adaptation3._CBMAAdaptation__vlan_config)
+
+    def test_creating_vlan_interface(self):
+        # VLAN interfaces has been created already
+        # when self.cbma_adaptation was instantiated.
+        # Try to create VLAN interfaces again with same settings. Should fail.
+        result = self.cbma_adaptation._CBMAAdaptation__create_vlan_interfaces()
+        self.assertFalse(result)
+
+    def test_deleting_vlan_interface(self):
+        # VLAN interfaces has been created already
+        # when CBMAAdaptation was instantiated for the first time.
+        # Try to delete them twice. Second deletion should fail.
+        result = self.cbma_adaptation._CBMAAdaptation__delete_vlan_interfaces()
+        self.assertTrue(result)
+        result = self.cbma_adaptation._CBMAAdaptation__delete_vlan_interfaces()
+        self.assertFalse(result)
+
+        # Finally create VLAN interfaces again to avoid confusing errors in
```

Comment:
I can't see the create VLAN phase mentioned in the comment.

Reply:
Fixed in commit bdd1774d11e99f5b28286a2ba5e4ebf16b9010a8. Comment itself is removed but vlan interfaces deletion is changed so that it doesn't try to delete non-existent interfaces that would cause somewhat confusing traces.